ext/native/thin3d/GLRenderManager.cpp

```diff
@@ -664,16 +686,17 @@ bool GLPushBuffer::AddBuffer() {
 	return true;
 }
 
-// Executed on the render thread!
 void GLPushBuffer::Destroy(bool onRenderThread) {
+	if (buf_ == -1)
+		return;  // Already destroyed
 	for (BufInfo &info : buffers_) {
 		// This will automatically unmap device memory, if needed.
 		// NOTE: We immediately delete the buffer, don't go through the deleter, if we're on the render thread.
 		if (onRenderThread) {
-			_dbg_assert_(G3D, OnRenderThread());
+			// _dbg_assert_(G3D, OnRenderThread());
 			delete info.buffer;
 		} else {
-			_dbg_assert_(G3D, !OnRenderThread());
+			// _dbg_assert_(G3D, !OnRenderThread());
```

Comment:
Hm, okay as long as the other thread is waiting.  I feel like it'd be nicer to keep the threads the same on shutdown...

Anyway, we should remove these rather than commenting them out.

-[Unknown]

Reply:
It's a bit unclear to me which thread the manual shutdown (main screen button) actually runs on ... should dig into that.